### PR TITLE
[bug](cooldown) Fix async_write_cooldown_meta and snapshot cooldowned version not continuous bug

### PR DESCRIPTION
**Pull request by platoneko**
## Proposed changes

Issue Number: close #xxx

1. Fix `async_write_cooldown_meta`
2. Fix snapshot cooldowned version not continuous bug

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
run buildall

File: be/src/olap/rowset/rowset_meta.h
Comment (airborne12) [inline be/src/olap/rowset/rowset_meta.h:1]: you get rid of resource_id function, but in const io::FileSystemSPtr& fs(), it is still being used

Comment (platoneko) [reply be/src/olap/rowset/rowset_meta.h:1]: fixed

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Comment by hello-stephen** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**